lib/github-pages-health-check/domain.rb

```diff
@@ -177,6 +178,18 @@ def apex_domain?
                             :ignore_private => true) == unicode_host
       end
 
+      # Does the domain have an SOA record published?
+      #
+      # Callers should be aware that this can return truthy for domains that
+      # are not apex-level (i.e. subdomain.apex.com).
+      def dns_zone_soa?
+        return @soa_records if defined?(@soa_records)
+        return false unless dns?
+
+        soa_records = dns.select { |answer| answer.type == Dnsruby::Types::SOA }
+        soa_records.any?
```

Comment:
Set `@soa_records` here? Or remove the reference to it at the beginning of the method?